api/pkg/imagebuilder/imagebuilder.go

```diff
@@ -43,6 +45,27 @@ import (
 
 var maxCheckImageRetry uint64 = 2
 
+var (
+	imageBuilderAttempts = promauto.NewCounterVec(prometheus.CounterOpts{
+		Namespace: "merlin",
+		Name:      "image_builder_attempts",
+		Help:      "Total of image builder attempts",
+	}, []string{"project", "model", "model_type", "model_version", "python_version"})
+
+	imageBuilderResults = promauto.NewCounterVec(prometheus.CounterOpts{
+		Namespace: "merlin",
+		Name:      "image_builder_results",
+		Help:      "Total of image builder results",
+	}, []string{"project", "model", "model_type", "model_version", "python_version", "result"})
+
+	imageBuilderDuration = promauto.NewHistogramVec(prometheus.HistogramOpts{
```

Comment:
Do you think the attempt # can also be a label in this metric instead of a separate `image_builder_attempts` metric? No strong opinion on this though.

Reply:
Attempt metrics should be separate counter metrics because they should be decoupled with the results of the image building process. If image building process fails, we still have the counter of the attempts. The ideal scenario is attempts == results, so if the count of attempts.

Also, having attempt metrics as label for duration metrics will double the size of duration metrics.